### PR TITLE
Remove builder dependency from the api.

### PR DESCRIPTION
**Pull request by calavera**
This change removes the builder dependency from the api.

I moved part of the build to the daemon and it's not responsible of creating the context and executing the build.

/cc @MHBauer, @icecrime 

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by MHBauer** [comment]
I am taking a similar approach to the runconfig stuff. I am not confident in if it is the right way to go. Do we really want to make the API something that just takes a reader stream and trusts the other side to correctly parse it? I feel like the parsing of any streams should definitely be done on the API side before giving it over. I am not sure.

This is how I would expect it to look.

File: docker/daemon.go [hunk 1]
```diff
@@ -413,7 +412,7 @@ func initRouter(s *apiserver.Server, d *daemon.Daemon) {
 		image.NewRouter(d, decoder),
 		systemrouter.NewRouter(d),
 		volume.NewRouter(d),
-		build.NewRouter(dockerfile.NewBuildManager(d)),
```
Comment (cpuguy83) [inline docker/daemon.go]: Looks like we can remove this `NewBuildManager` function.

**Comment by cpuguy83** [comment]
@MHBauer I think this approach makes since for this case. The stream is the context.

File: docker/daemon.go (hunk 1 above)
Comment (calavera) [inline docker/daemon.go]: it's still used in https://github.com/docker/docker/pull/21636/files#diff-9466c26908f68c0d8d73c7a0e23fae9dR29

Comment (cpuguy83) [inline docker/daemon.go]: gah! Protip: checkout the branch with the changes before doing `git grep`

**Comment by cpuguy83** [comment]
LGTM

File: docker/daemon.go [hunk 2]
```diff
@@ -413,7 +412,7 @@ func initRouter(s *apiserver.Server, d *daemon.Daemon) {
...
+		build.NewRouter(d),
```
Comment (MHBauer) [inline docker/daemon.go]: This is probably the sweetest line for me. We're at all the routers taking a straight `backend`, with no ties. It's nice.

**Comment by MHBauer** [comment]
If you feel comfortable going this way. I'll accept it. 

LGTM.

**Pull request merged**